src/gmail.js

```diff
@@ -1885,15 +1906,18 @@ var Gmail = function(localJQuery) {
 
             // only append email to cache if not already there.
             if (thread.emails.filter(i => i.id === email.id).length === 0) {
-                //console.log("push",email);
+                //console.log("append email to cache",data_source, email) ;
                 thread.emails.push(email);
             }
             else
             {
-                let index = thread.emails.findIndex(i => i.id === email.id);
-                //console.log("before:",thread.emails[index]);
-                thread.emails[index] = email;
-                //console.log("after:",thread.emails[index]);
+                // Only update cache with data source fd_request_payload and fd_embedded_json
+                if (data_source === "fd_request_payload" || data_source === "fd_embedded_json") {
```

Comment:
This could be expressed as an `else if` to reduce nesting.

Reply:
I agree, done